### PR TITLE
Fix bytecode flags

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
Resolves  https://github.com/brave/brave-browser/issues/10569

Fixes asm bytecode patching for ExternalNavigationHandler and AndroidSyncSettings classes

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
See https://github.com/brave/brave-browser/issues/10569

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by AlexeyBarabash** [comment]
Skipped CI all except Android  because PR has only patches for BUILD.gn around the Java files.

File: patches/components-external_intents-android-BUILD.gn.patch [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
-index 530cfa2c449b94854ace00cc710818fb496d2a7e..4dab931ff4d955f8a04bfe7bdfce7cde9f37ebd8 100644
+index 530cfa2c449b94854ace00cc710818fb496d2a7e..6d06fdb8448349621a86d0c9ff5f0d88acbe0eac 100644
 --- a/components/external_intents/android/BUILD.gn
 +++ b/components/external_intents/android/BUILD.gn
-@@ -57,6 +57,7 @@ static_library("android") {
+@@ -29,6 +29,7 @@ android_library("java") {
+     "//ui/android:ui_java",
+     "//url:gurl_java",
+   ]
++  deps += brave_chrome_java_deps enable_bytecode_rewriter = true
```
Comment (samartnik) [inline patches/components-external_intents-android-BUILD.gn.patch:9]: We don't need `deps` here, only `enable_bytecode_rewriter`. Also, I think we should find one place where we can patch it for all the components if it's possible.

File: patches/components-sync-android-BUILD.gn.patch [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
-index 4575d881584549f4de1afd09d226002428149ca9..65a143eacd1f2b94eddf5b92cc110f1eeb25af07 100644
+index 4575d881584549f4de1afd09d226002428149ca9..ba5805a71d055e4c7def67baad2f050300a6ae6d 100644
 --- a/components/sync/android/BUILD.gn
 +++ b/components/sync/android/BUILD.gn
-@@ -32,6 +32,7 @@ android_library("sync_java") {
+@@ -17,6 +17,7 @@ android_library("sync_java") {
+     "//third_party/cacheinvalidation:cacheinvalidation_javalib",
+     "//third_party/cacheinvalidation:cacheinvalidation_proto_java",
+   ]
++  deps += brave_chrome_java_deps enable_bytecode_rewriter = true
```
Comment (samartnik) [inline patches/components-sync-android-BUILD.gn.patch:9]: Same here

File: patches/components-external_intents-android-BUILD.gn.patch (hunk 1 above)
Comment (AlexeyBarabash) [reply patches/components-external_intents-android-BUILD.gn.patch]: @samartnik  fixed at https://github.com/brave/brave-core/compare/fb01c0a448cb7a75dbb9729ec2c3b32d6c9f2c00..44239ddb3f5c8b79743cfa502297750261ae8f5b

File: patches/components-sync-android-BUILD.gn.patch (hunk 2 above)
Comment (AlexeyBarabash) [reply patches/components-sync-android-BUILD.gn.patch]: @samartnik  fixed at https://github.com/brave/brave-core/compare/fb01c0a448cb7a75dbb9729ec2c3b32d6c9f2c00..44239ddb3f5c8b79743cfa502297750261ae8f5b

**Comment by AlexeyBarabash** [comment]
@brianjohnson, 
me and @samartnik tried to find the common point to set variable `enable_bytecode_rewriter = true` to avoid patching of each interested `BUILD.gn`.
Patch 
```
--- a/build/config/android/rules.gni
+++ b/build/config/android/rules.gni
                             [
@@ -1841,6 +1843,7 @@ if (enable_java_templates) {
 
       supports_android = true
       requires_android = true
+      enable_bytecode_rewriter = true
 
       if (!defined(jar_excluded_patterns)) {
         jar_excluded_patterns = []
```

gave errors.

And patch attempts near `_enable_bytecode_rewriter` https://source.chromium.org/chromium/chromium/src/+/refs/tags/84.0.4147.68:build/config/android/internal_rules.gni;l=1682 of forcing it to true, wrapping with conditions - gave either errors, or bytecode wasn't actually patched.

Maybe you have an idea how to set `enable_bytecode_rewriter` at the one place for all java libs?

**Comment by AlexeyBarabash** [comment]
CI failed for audit-deps, rebasing.

**Comment by AlexeyBarabash** [comment]
CI failed on `audit-deps` , this is known issue on all platforms

**Comment by AlexeyBarabash** [comment]
Decreased lines number at patches/components-sync-android-BUILD.gn.patch (https://github.com/brave/brave-core/compare/eec6f00dbd8f99e69dfadca1a305591889c8aa03..d9381d08e8cc2de961246201dcc81074d2ab8ba7)

**Comment by AlexeyBarabash** [comment]
CI still fails with a known audit deps issue

**Pull request merged**